src/Features/CSharp/Portable/CodeFixes/HideBase/HideBaseCodeFixProvider.RemoveNewKeywordAction.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.HideBase
+{
+    internal partial class HideBaseCodeFixProvider
+    {
+        private class RemoveNewKeywordAction : CodeActions.CodeAction
+        {
+            private readonly Document _document;
+            private readonly SyntaxNode _node;
+
+            public override string Title => CSharpFeaturesResources.Remove_new_keyword;
+
+            public RemoveNewKeywordAction(Document document, SyntaxNode node)
+            {
+                _document = document;
+                _node = node;
+            }
+
+            protected override async Task<Document> GetChangedDocumentAsync(CancellationToken cancellationToken)
+            {
+                var syntaxFacts = CSharpSyntaxFacts.Instance;
+
+                var newModifier = GetNewModifier(_node);
+                Debug.Assert(newModifier != default, $"'new' keyword was not found, but diagnostic was triggered");
+
+                var newNode = _node;
+
+                if (newModifier.HasTrailingTrivia || newModifier.HasLeadingTrivia)
+                {
+                    var newModifierTrivia = newModifier.GetAllTrivia().ToSyntaxTriviaList();
+                    var previousToken = newModifier.GetPreviousToken();
+                    var nextToken = newModifier.GetNextToken();
+
+                    var sourceText = _document.GetTextSynchronously(cancellationToken);
+                    var isFirstTokenOnLine = newModifier.IsFirstTokenOnLine(sourceText);
+
+                    var newTrivia = new SyntaxTriviaList();
+                    if (!isFirstTokenOnLine)
+                        newTrivia = newTrivia.AddRange(previousToken.TrailingTrivia);
+                    newTrivia = newTrivia
+                        .AddRange(newModifierTrivia)
+                        .AddRange(nextToken.LeadingTrivia)
+                        .CollapseSequentialWhitespaceTrivia();
+
+                    if (isFirstTokenOnLine)
+                    {
+                        var nextTokenWithMovedTrivia = nextToken.WithLeadingTrivia(newTrivia);
+                        newNode = newNode.ReplaceToken(nextToken, nextTokenWithMovedTrivia);
+                    }
+                    else
+                    {
+                        var previousTokenWithMovedTrivia = previousToken.WithTrailingTrivia(newTrivia);
+                        newNode = newNode.ReplaceToken(previousToken, previousTokenWithMovedTrivia);
+                    }
+                }
```

Comment:
a much simpler approach would be:

1. grab all the trivia on the `new` modifier`
2. attach it to the previous token as trailing trivia
3. add an ElasticTrivia to the previous token's trailing trivia as well.

This should ensure the formatter cleans up things like multiple-whitespaces between the previous and next token.

Reply:
I tried this approach and faced two issues:
1. If `new` is the first token on the line, attaching trivia to previous token seems to be not working at all, it got removed completely. Failed test:
```
 class App
 {
-    /* start */ /* end */ int Test;
+int Test;
 }
```
2. Formatter starts inserting line breaks after multiline comments in elastic trivia. Failed test:
```
 class App
 {
-    /* start */ public /* middle */ /* end */ int Test;
+    /* start */
+    public /* middle */     /* end */ int Test;
 }
```

I thought that I should preserve original code structure as much as possible and reduce any side effects. If I'm on the wrong way, I'd be happy to revisit my hacky approach.

Reply:
> If new is the first token on the line, attaching trivia to previous token seems to be not working at all, it got removed completely. Failed test:

something is strange there.  i'd need to see what your code was doing.

Reply:
note: given that 'new' might be first, it's prob best to just attach to next token.  but still add elastic so that multiple whitespace is cleaned up.

Reply:
Thanks, moving to the next token worked better, however, there are still some failing tests due to extra line breaks/spaces inserted. I pushed the code to temp branch: https://github.com/vanashimko/roslyn/commit/4c32b00bcbca08dca21aebb89460357867293391.

Failed tests:
1) 
```
 class App
 {
-    public const int test = 1;
+    public const  int test = 1;
 }
```
2) 
```
 class App
 {
-    /* start */ /* end */ int Test;
+    /* start */
+    /* end */
+    int Test;
 }
```
3) 
```
 class App
 {
-    /* start */ public /* middle */ /* end */ int Test;
+    /* start */ public /* middle */     /* end */ int Test;
 }
```
4) 
 ```
 class App
 {
-    /* start */ public /* middle */ /* end */ int Test;
+    /* start */ public /* middle */  /* end */ int Test;
 }
```
5) 
```
 class App
 {
-    /* end */ int Test;
+    /* end */
+    int Test;
 }
```

Reply:
JFYI if somebody is interested, discussion is in progress in the actual commit in another branch: https://github.com/vanashimko/roslyn/commit/4c32b00bcbca08dca21aebb89460357867293391#r38914037